pkg/fileservice/local_fs.go

```diff
@@ -528,6 +528,11 @@ func (l *LocalFS) read(ctx context.Context, vector *IOVector, bytesCounter *atom
 			if err != nil {
 				return err
 			}
+			defer func() {
+				if err != nil {
```

Comment:
why defer file.Close() is not enough here?

Reply:
if ReadCloserForRead is not nil, the caller is requesting an io.ReadCloser, so we return a wrapped `*os.File`. The returning file will be closed when the caller calls Close, not before Read returns.